src/features/joinForms/components/JoinFormList.tsx

```diff
@@ -2,21 +2,44 @@ import { FC } from 'react';
 import { Box, Card, Divider } from '@mui/material';
 
 import JoinFormListItem from './JoinFormListItem';
+import messageIds from '../l10n/messageIds';
+import { useMessages } from 'core/i18n';
 import { ZetkinJoinForm } from '../types';
+import ZUIEllipsisMenu from 'zui/ZUIEllipsisMenu';
+import ZUIInlineCopyToClipboard from 'zui/ZUIInlineCopyToClipBoard';
 
 type Props = {
   forms: ZetkinJoinForm[];
   onItemClick: (form: ZetkinJoinForm) => void;
 };
 
 const JoinFormList: FC<Props> = ({ forms, onItemClick }) => {
+  const messages = useMessages(messageIds);
+
   return (
     <Box m={2}>
       <Card>
         {forms.map((form, index) => (
           <Box key={form.id}>
             {index > 0 && <Divider />}
-            <JoinFormListItem form={form} onClick={() => onItemClick(form)} />
+            <Box display="flex" justifyContent="space-between">
+              <JoinFormListItem form={form} onClick={() => onItemClick(form)} />
+              <ZUIEllipsisMenu
+                items={[
+                  {
+                    label: (
+                      <ZUIInlineCopyToClipboard
+                        alwaysShowIcon
+                        clickableChildren
+                        copyText={form.submit_token}
+                      >
+                        {messages.formList.copyToken()}
+                      </ZUIInlineCopyToClipboard>
+                    ),
+                  },
```

Comment:
Instead of rendering a `ZUIInlineCopyToClipboard` inside of the ellipsis menu, I think we should just add an `onSelect` that does the same thing that the `ZUIInlineCopyToClipboard` component would do, but without all of the extra fuzz.

```suggestion
                  {
                    label: messages.formList.copyToken(),
                    onSelect: () => {
                        copy(form.submit_token);
                        showSnackbar(
                          'success',
                          messages.copyToClipboard.copiedValue({
                            value: str.length > 30 ? str.slice(0, 30) + '…' : str,
                          })
                        );
                    },
                  },
```

This also requires some additional changes to this component/module of course, like getting `showSnackbar()` from the context, importing `copy()` and getting/adding the relevant message.